pkg/auth/spire/certificate_provider_test.go

```diff
@@ -389,8 +390,24 @@ func TestSpireDelegateClient_GetCertificateForIdentity(t *testing.T) {
 				t.Errorf("SpireDelegateClient.GetCertificateForIdentity() error = %v, wantErr %v", err, tt.wantErr)
 				return
 			}
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("SpireDelegateClient.GetCertificateForIdentity() = %v, want %v", got, tt.want)
+			if tt.want != nil {
+				if !assert.Equal(t, tt.want.Certificate, got.Certificate) {
+					t.Errorf("SpireDelegateClient.GetCertificateForIdentity().Certificate = %v, want %v", got.Certificate, tt.want.Certificate)
+				}
+				wantKey := tt.want.PrivateKey.(*rsa.PrivateKey)
+				gotKey := got.PrivateKey.(*rsa.PrivateKey)
+				if !wantKey.Equal(gotKey) {
```

Comment:
is it really necessary to have this additional equality check on the type itself - even though we have the following equals assertion?

IMO if one of them isn't reliable we should simply not use it for assertion at all.